lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
@dtzWill Yeah, thanks, I did that and still had some serious issues...  Some of it, I think, is just the mess of Xcode and multiple llvm builds.  Also I think some of the expected directories coming out of the build I had renamed... eh.. it's a new machine, I kind of dumped some of the old apps/tools in and added some new and haven't quite had time to sort the cruft, but I'll get there.

I'm curious what you mean regarding your last paragraph.  Are you referring to building a CFG based model to dynamically check pointers?  Something akin to CFI (Control Flow Integrity) but with more analytical depth?  Do you care how the source might be altered/appended to, something like Seahorn or Pagai?  Anyway, just genuinely curious as to what you're after there...

As this is starting to get off tangent, perhaps email is best, but whatever works...  Thanks again.